cpp/benchmarks/wavelet_operator_mpi.cc

```diff
@@ -48,93 +50,65 @@ using namespace purify;
-      std::make_tuple("DB3", 3u),   std::make_tuple("DB4", 3u), std::make_tuple("DB5", 3u),
-      std::make_tuple("DB6", 3u),   std::make_tuple("DB7", 3u), std::make_tuple("DB8", 3u)};
-
-  sopt::wavelets::SARA const saraDistr = sopt::wavelets::distribute_sara(m_sara, m_world);
-
-  sopt::LinearTransform<Vector<t_complex>> m_Psi =
-      sopt::linear_transform<t_complex>(saraDistr, m_imsizey, m_imsizex, m_world);
-
+BENCHMARK_DEFINE_F(WaveletOperatorMPIFixture, Forward)(benchmark::State& state) {
   // Benchmark the application of the operator
 
-  // Get the number of wavelet coefs
-  t_uint const n_wave_coeff = saraDistr.size() * m_imsizey * m_imsizex;
-
   // Apply Psi to a temporary vector
   Vector<t_complex> image = Vector<t_complex>::Random(m_imsizey * m_imsizex);
   Vector<t_complex> const wavelet_coeff = Vector<t_complex>::Ones(n_wave_coeff);
 
-  while (state.KeepRunning()) {
+  while (m_world.broadcast<int>(state.KeepRunning())) {
     auto start = std::chrono::high_resolution_clock::now();
     image = m_Psi * wavelet_coeff;
     auto end = std::chrono::high_resolution_clock::now();
-    state.SetIterationTime(b_utilities::duration(start, end));
+    state.SetIterationTime(b_utilities::duration(start, end, m_world));
   }
 }
 
-BENCHMARK_DEFINE_F(WaveletOperatorAdjointMPIFixture, Apply)(benchmark::State& state) {
-  // Image size
-  t_uint m_imsizex = state.range(0);
-  t_uint m_imsizey = state.range(0);
-
-  sopt::wavelets::SARA m_sara{
-      std::make_tuple("Dirac", 3u), std::make_tuple("DB1", 3u), std::make_tuple("DB2", 3u),
-      std::make_tuple("DB3", 3u),   std::make_tuple("DB4", 3u), std::make_tuple("DB5", 3u),
-      std::make_tuple("DB6", 3u),   std::make_tuple("DB7", 3u), std::make_tuple("DB8", 3u)};
-
-  sopt::wavelets::SARA const saraDistr = sopt::wavelets::distribute_sara(m_sara, m_world);
-
-  sopt::LinearTransform<Vector<t_complex>> m_Psi =
-      sopt::linear_transform<t_complex>(saraDistr, m_imsizey, m_imsizex, m_world);
-
-  // Benchmark the application of the operator
-
-  // Get the number of wavelet coefs
-  t_uint const n_wave_coeff = saraDistr.size() * m_imsizey * m_imsizex;
-
+BENCHMARK_DEFINE_F(WaveletOperatorMPIFixture, Adjoint)(benchmark::State& state) {
   // Apply Psi to a temporary vector
   Vector<t_complex> const image = Vector<t_complex>::Ones(m_imsizey * m_imsizex);
   Vector<t_complex> wavelet_coeff = Vector<t_complex>::Zero(n_wave_coeff);
 
-  while (state.KeepRunning()) {
+  while (m_world.broadcast<int>(state.KeepRunning())) {
     auto start = std::chrono::high_resolution_clock::now();
     wavelet_coeff = m_Psi.adjoint() * image;
     auto end = std::chrono::high_resolution_clock::now();
-    state.SetIterationTime(b_utilities::duration(start, end));
+    state.SetIterationTime(b_utilities::duration(start, end, m_world));
   }
 }
 
-BENCHMARK_REGISTER_F(WaveletOperatorMPIFixture, Apply)
+BENCHMARK_REGISTER_F(WaveletOperatorMPIFixture, Forward)
```

Comment:
OK then you just have to comment out the one you don't want to run. Makes sense to me. That's indeed much better. Good job